### DISCUSSION TITLE
What properties are possible in each attribute for each element?

### DISCUSSION THREAD

**Discussion by paul23-git** [clientIO/joint#2842] [Q&A]
### Introduction

The tutorial has a lot of explanation how the attr elements work.

However it never described *which* deeper settings are possible. IE I know a link has the selector "label", but it never explained that through the label I can give a text. (well that example is often used, but I cannot find a definite list of what's possible.

Where would I find that, is there a formal source?

Or are they all the same (line, label, root etc all share the same property list?)

### Steps to reproduce

not relevant

### Restrictions & Constraints

not relevant, documentation.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment]
Let's take a look at `standard.Rectangle` for example,

According to the [documentation](https://docs.jointjs.com/api/shapes/standard/Rectangle), the available selectors are:

selector|type|description
-- | -- | --
root | SVGGElement | Container of all nodes
body | SVGRectElement | Rectangular body of the shape
label | SVGTextElement | Text inside the body

The SVG markup looks like this:

```svg
<g @selector="root">
  <rect @selector="body"/>
  <text @selector="label"/>
</g>
```

For example, the `label` selector targets an [SVGTextElement](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/text). You can apply any SVG attributes compatible with the `<text>` SVG element.
_e.g._ `el.attr('label/fill', 'red');`

Additionally, you can utilize SVG attributes specific to the JointJS context: https://docs.jointjs.com/api/dia/attributes/
_e.g._ `el.attr('label/text', 'Hello!');`

**Reply by paul23-git** [reply]
Oh thank you, I do notice some discrepancies though, say the link element, the documentation says the root is an SVGGElement (https://docs.jointjs.com/api/shapes/standard/Link/).

Yet typescript and the typing in code says it's an SVGAttribute:

![image](https://github.com/user-attachments/assets/53b1f045-a5cd-4c29-a0cc-9e9ed9d11763)


What is the stronger guide?

**Reply by kumilingus** [reply]
In this case, it is the documentation that is more accurate.
The `root` selector always points to a `<g/>` element.
